airflow/api/common/experimental/mark_tasks.py

```diff
@@ -62,8 +62,10 @@ def _create_dagruns(dag, execution_dates, state, run_type):
 
 @provide_session
 def set_state(
+    *,
     tasks: Iterable[BaseOperator],
-    execution_date: datetime.datetime,
+    execution_date: Optional[datetime.datetime] = None,
+    run_id: Optional[str] = None,
```

Comment:
Should we check only _exactly one_ of `run_id` and `execution_date` are passed to this function? If not, we should mention in the docstring this function expects the _caller_ to check first.

Reply:
Done